naga/src/front/wgsl/parse/directive/enable_extension.rs

```diff
@@ -0,0 +1,112 @@
+//! `enable …;` extensions in WGSL.
+//!
+//! The focal point of this module is the [`EnableExtension`] API.
+use crate::{front::wgsl::error::Error, Span};
+
+/// Tracks the status of every enable extension known to Naga.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct EnableExtensions {}
+
+impl EnableExtensions {
+    pub(crate) const fn empty() -> Self {
+        Self {}
+    }
+
+    /// Add an enable extension to the set requested by a module.
+    #[allow(unreachable_code)]
+    pub(crate) fn add(&mut self, ext: ImplementedEnableExtension) {
+        let _field: &mut bool = match ext {};
+        *_field = true;
```

Comment:
`match`ing on an uninhabited `enum` actually returns `!`, which can be coerced to any type. The explicit type does the coercion, instead of letting it fall back to `()`.

Reply:
As a reader of the code I find that surprising, do we need this code at all? I guess it doesn't really matter either way since the next PR will add the proper logic here right?

Reply:
@teoxoy: Yeah, the intent was to minimize the diff. between here and the next PR. However, it doesn't look like F16 extension support is going to land for a while, based on the momentum I perceive. We _could_ simply remove the body of this function for now, but 🤷🏻‍♂️ I honestly just consider it a logistic detail, at this point.

We could also add a test-only variant to ensure that this works as intended, but IDK if it's worth the effort over "just" implementing a real extension.

Reply:
I would prefer we remove the body but up to you.